### PR TITLE
Fix the clippy::uneeded_field_pattern lint, enable it

### LINKED ISSUE #166: Fix failures for lint clippy::unneeded_field_pattern
# Clippy reports for lint clippy::unneeded_field_pattern

Issues found:
<details>
<summary>
⚠️ you matched a field with a wildcard pattern, consider using `..` instead
 -- <a href="https://github.com/JustusFluegel/rust-ga/blob/8576fd7da643fd30f1e64b33d51b29a2afb7b063/packages/push/examples/simple_regression/main.rs#L108C9-L108C29">packages/push/examples/simple_regression/main.rs</a>
</summary>

```txt
warning: you matched a field with a wildcard pattern, consider using `..` instead
   --> packages/push/examples/simple_regression/main.rs:108:9
    |
108 |         max_genome_length: _,
    |         ^^^^^^^^^^^^^^^^^^^^
    |
    = help: try with `Args { run_model, population_size, max_initial_instructions, num_generations, .. }`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unneeded_field_pattern
    = note: requested on the command line with `-W clippy::unneeded-field-pattern`
```

</details>
<details>
<summary>
⚠️ you matched a field with a wildcard pattern, consider using `..` instead
 -- <a href="https://github.com/JustusFluegel/rust-ga/blob/8576fd7da643fd30f1e64b33d51b29a2afb7b063/packages/push/examples/complex_regression/main.rs#L108C9-L108C29">packages/push/examples/complex_regression/main.rs</a>
</summary>

```txt
warning: you matched a field with a wildcard pattern, consider using `..` instead
   --> packages/push/examples/complex_regression/main.rs:108:9
    |
108 |         max_genome_length: _,
    |         ^^^^^^^^^^^^^^^^^^^^
    |
    = help: try with `Args { run_model, population_size, max_initial_instructions, num_generations, .. }`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unneeded_field_pattern
    = note: requested on the command line with `-W clippy::unneeded-field-pattern`
```

</details>

### PR DESCRIPTION
**Pull request by JustusFluegel**
Closes #166

### CODE REVIEW

**Review by NicMcPhee — APPROVED** [review]
Nice – thanks!

I need to be more aware of `..` than I am, so it's nice that both you and Clippy are here to keep me honest. 😜

File: Cargo.toml [hunk 1]
```diff
@@ -62,8 +62,13 @@ try_err = "warn"
 unnecessary_safety_comment = "warn"
 unnecessary_safety_doc = "warn"
 unnecessary_self_imports = "warn"
+
 unneeded_field_pattern = "warn"
-# use_debug = "warn"
+
+
+
```
Comment (NicMcPhee) [inline Cargo.toml:69]:
```suggestion



```

We don't need this white space, do we?

File: Cargo.toml [hunk 2]
```diff
@@ -86,7 +91,11 @@ allow_attributes = "deny"
 cargo_common_metadata = "allow"
 redundant_type_annotations = "allow"
 todo = "allow"
-panic = "allow"
+
+
+
+
+
```
Comment (NicMcPhee) [inline Cargo.toml:98]:
```suggestion





```

And again.

**Review by NicMcPhee — CHANGES_REQUESTED** [review]

**Comment by JustusFluegel** [comment]
And again. Sorry about that.

**Review by NicMcPhee — APPROVED** [review]

**Pull request merged**